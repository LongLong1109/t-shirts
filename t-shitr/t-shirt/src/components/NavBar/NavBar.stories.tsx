import type { Meta, StoryObj } from '@storybook/react'
 
// components
import NavBar from '@/components/NavBar'

const meta = {
  title: 'Components/NavBar',
  component: NavBar,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof NavBar>

export default meta
type Story = StoryObj<typeof NavBar>

export const NavBarPrimary: Story = {
  args: {},
}
