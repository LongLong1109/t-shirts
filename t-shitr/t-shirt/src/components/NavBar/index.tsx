'use client'
import { memo } from 'react'
import isEqual from 'react-fast-compare'

import DotIcon from '@/icons/dotIcon'
import Link from 'next/link'
import { usePathname } from 'next/navigation'

// mocks
import { navBarListMock } from '@/mocks/navBarList'

const NavBar = () => {
  const pathname = usePathname()

  return (
    <nav className='flex sm:bg-white-100 sm:w-full'>
      <ul className='flex md:flex-row md:p-0 gap-10 sm:flex-col sm:p-10'>
        {navBarListMock.map(({ url, name }, index) => {
          const isActive = pathname.includes(url)
          return (
            <li key={index}>
              <Link
                href={url}
                className={`${isActive ? 'active font-bold flex flex-col items-center' : ''} text-sm`}
                style={{ color: isActive ? 'blue' : 'black' }}
              >
                <p>{name}</p>
                {isActive && <DotIcon />}
              </Link>
            </li>
          )
        })}
      </ul>
    </nav>
  )
}

export default memo(NavBar, isEqual)
