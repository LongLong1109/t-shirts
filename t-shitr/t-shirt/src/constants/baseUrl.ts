export const BASE_URL = process.env.NEXT_PUBLIC_URL
export const PRODUCT_URL = process.env.NEXT_PUBLIC_PRODUCT
export const CART_URL = process.env.NEXT_PUBLIC_CART
