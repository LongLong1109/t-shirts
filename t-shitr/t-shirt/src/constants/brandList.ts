import { basset, furniture, lazBoy, blackHawk, levizt } from '@/assets/images'

export const brandList = [
  { src: basset, alt: 'basset' },
  { src: furniture, alt: 'furniture' },
  { src: lazBoy, alt: 'lazBoy' },
  { src: blackHawk, alt: 'blackHawk' },
  { src: levizt, alt: 'levizt' },
]
