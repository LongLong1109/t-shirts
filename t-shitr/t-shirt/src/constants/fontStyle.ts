export const FONT_SIZE = {
  XS: '14px',
  SM: '18px',
  MD: '40px',
  LG: '48px',
}

export const FONT_WEIGHT = {
  THIN: 'thin',
  HAIR_LINE: 'hairline',
  LIGHT: 'light',
  NORMAL: 'normal',
  MEDIUM: 'medium',
  SEMI_BOLD: 'semibold',
  BOLD: 'bold',
  EXTRA_BOLD: 'extrabold',
  BLACK: 'black',
}

export type FontSize = (typeof FONT_SIZE)[keyof typeof FONT_SIZE]
export type FontWeight = (typeof FONT_WEIGHT)[keyof typeof FONT_WEIGHT]
