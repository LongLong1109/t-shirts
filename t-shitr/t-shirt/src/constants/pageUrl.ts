export const PAGE_URL = {
  SIGN_IN: '/login',
  SIGN_UP: 'signUp',
  HOME: '/home',
  MY_ACCOUNT: '/my-account',
  ABOUT_US: '/about-us',
  CONTACT: '/contact',
  PRODUCT_DETAIL: '/productDetail',
  SHOP: '',
  CART: '/cart',
}
