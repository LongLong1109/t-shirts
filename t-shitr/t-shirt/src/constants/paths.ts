export const PATH_NAME = {
  HOME: 'Home',
  LOGIN: 'Login',
  SIGN_UP: 'Sign Up',
  MY_ACCOUNT: 'My Account',
  ABOUT_US: 'About Us',
  CONTACT: 'Contact',
  SHOP: 'Shop',
  CART: 'Cart',
}
