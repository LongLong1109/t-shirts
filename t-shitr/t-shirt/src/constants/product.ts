// interface
import { ProductProps } from '@/interfaces/product'

export const INITIAL_PRODUCT: ProductProps = {
  id: '',
  name: '',
  image: '',
  price: '',
  color: '',
  size: '',
  inStock: '',
  quantity: 1,
  blurDataURL: '',
}

export const PRODUCT_IMAGES = [
  {
    src: 'https://s3-alpha-sig.figma.com/img/4cac/1c46/34e9b98b924cfd8fd8d82561815b9ba6?Expires=1722816000&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=UjpZI3n-ctvOxZA-vRbJ5oIJCPbxovMSFAHD1KZ3Iy~87Ix7W3YkE4WWR9HOCt0d6TW5~P69RrX0uk~blW-~b3N2J2N2ZYDpA3bJcHQtoRr-ewHhz7fM57kXZsMBLZFl-Vl40Kp6y0GQuzOBo2YIKeFY0Zu9OZgtqmg3KX7wsw7m0nuuCoi1tHGH8XFvhSP-aefHS5QqhDHe0~ALls-B0EgkaKjgu2jGYVhAfaE2U1e~ie67HY3NRUl4quNuN0eoJ6cNwhK0KRQIRwMxt-uflH6q1KZp9rhLWu2fC2prdT8zc53WvSc4QdaUHML2xld4cLv-tPg8-Uq5EA6N4uhW5w__',
    alt: 'White T-Shirt',
    blurData:
      'Imh0dHBzOi8vYml6d2ViLmRrdGNkbi5uZXQvdGh1bWIvMTAyNHgxMDI0LzEwMC80MzgvOTI2L3Byb2R1Y3RzLzc4OTQ3ODI1NTQwMjI0MDYwMTU2NjM3NjY3NDc3OS5wbmc/dj0xNjMzNTc3ODg3MzcwIg==',
  },
  {
    src: 'https://s3-alpha-sig.figma.com/img/acbb/0249/a1432b26fadbdae44b7595dff2bf552b?Expires=1722816000&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=QeHasBekOFrYbzy7etchog-1Uw328D5EeP4aDU6JBSuoGGlSWUcVvm4dn4UyhHYExQSlv2QN7peCj9gja989rlNMv9xxDAVwG58hwejs6x1~z1WpY9jNRwLjKiIzUuDbOenEWxUy3K7P~4LnoKtW45Shs1J1fz~nahdgcnuO9xUDr9aOOsKypE6dtQfssZpD7RGeD9s~Pq7p5XDb~mbqTNBsQ8HrDHkNYYQ-FNcMted7prvxB-FFr-OyAG-DfLu588624BG9z8SJfxJbyBNpp6pzmumI9BZ-d24PFJelnLiXqiWcHF2pRz5flfyoKvGcmFBaTpqIbPWYjzjMnElnAA__',
    alt: 'White T-Shirt',
    blurData:
      'Imh0dHBzOi8vYml6d2ViLmRrdGNkbi5uZXQvdGh1bWIvMTAyNHgxMDI0LzEwMC80MzgvOTI2L3Byb2R1Y3RzLzc4OTQ3ODI1NTQwMjI0MDYwMTU2NjM3NjY3NDc3OS5wbmc/dj0xNjMzNTc3ODg3MzcwIg==',
  },
  {
    src: 'https://s3-alpha-sig.figma.com/img/f1c8/187f/5de17f4586499f62c3a89ae9fb05327c?Expires=1722816000&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4&Signature=KOpK14ahjzbl6bYej01DXh6KjZsm1Tfpx58RxH053fI9jORVv6IPKvLkTzxyQOnO8V9sbshNRQ0L4-O8ccP0iXu48Nl7X47NKNQQ3t~I60dzDJ-DMAjt7RGl9KdkjnqpPfz9~Wonjhifwzd5U7v400VVtvXrG3lRxYahMDspFlTSNtsQ-SH9mg6cgY7kkbfO5UF9g8LM9gcNK9NkDJdbd25U1pkf82wzQ3PildSceWVpEw6bkmxCfL0DRBADblhVvJ-g77KYCS7orb3momx88hltfnGivn0BHddgEcyJ~9BqJ6tVTP~LsYJAhiS0YhnFvnl4Jq1gHZ~3oaP0k-txdw__',
    alt: 'White T-Shirt',
    blurData:
      'Imh0dHBzOi8vYml6d2ViLmRrdGNkbi5uZXQvdGh1bWIvMTAyNHgxMDI0LzEwMC80MzgvOTI2L3Byb2R1Y3RzLzc4OTQ3ODI1NTQwMjI0MDYwMTU2NjM3NjY3NDc3OS5wbmc/dj0xNjMzNTc3ODg3MzcwIg==',
  },
]
