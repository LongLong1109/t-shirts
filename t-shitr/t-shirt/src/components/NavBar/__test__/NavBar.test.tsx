import { render, screen } from '@testing-library/react';
import { usePathname } from 'next/navigation';

// components
import NavBar from '@/components/NavBar';

// constants
import { PAGE_URL } from '@/constants/pageUrl';

jest.mock('next/navigation', () => ({
  usePathname: jest.fn(),
}));

describe('NavBar Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should matches snapshot', () => {
    (usePathname as jest.Mock).mockReturnValue(PAGE_URL.HOME);
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });

  it('renders NavBar with provided navBarList', () => {
    (usePathname as jest.Mock).mockReturnValue(PAGE_URL.HOME);
    render(<NavBar />);

    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument();
  });

  it('does not apply active class and styles to inactive items', () => {
    (usePathname as jest.Mock).mockReturnValue(PAGE_URL.HOME);
    render(<NavBar />);

    const inactiveItem = screen.getByText('About');
    expect(inactiveItem).not.toHaveClass('active');
  });
});
