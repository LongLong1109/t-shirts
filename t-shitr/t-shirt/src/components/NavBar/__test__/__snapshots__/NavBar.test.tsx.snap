// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should matches snapshot 1`] = `
<div>
  <nav
    class="flex sm:bg-white-100 sm:w-full"
  >
    <ul
      class="flex md:flex-row md:p-0 gap-10 sm:flex-col sm:p-10"
    >
      <li>
        <a
          class="active font-bold flex flex-col items-center text-sm"
          href="/home"
          style="color: blue;"
        >
          <p>
            Home
          </p>
          <svg
            fill="none"
            height="4"
            viewBox="0 0 4 4"
            width="4"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="2"
              cy="2"
              fill="#5463FF"
              r="2"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class=" text-sm"
          href="/shop"
          style="color: black;"
        >
          <p>
            Shop
          </p>
        </a>
      </li>
      <li>
        <a
          class=" text-sm"
          href="/about-us"
          style="color: black;"
        >
          <p>
            About
          </p>
        </a>
      </li>
      <li>
        <a
          class=" text-sm"
          href="/contact"
          style="color: black;"
        >
          <p>
            Contact
          </p>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
